### DISCUSSION TITLE
After elimination of noise, mmocr still can't give correct result

### DISCUSSION THREAD

**Discussion by melgino** [open-mmlab/mmocr#1590] [Q&A]
Previous picutre:
![yehaibo_ocr_001](https://user-images.githubusercontent.com/537730/205286303-04c48f26-33d9-4f7a-b9e8-1ec14d5fa826.jpg)
After elimination of noise
![222](https://user-images.githubusercontent.com/537730/205286343-adbd1a33-79fd-4242-9bdf-0a4701aa278e.jpg)

I install mmocr according to github install guide.
after i execute:
`python mmocr/utils/ocr.py D:/screenshot/222.jpg --output demo/ --print-result --imshow`
![out_222](https://user-images.githubusercontent.com/537730/205286896-ed3514fe-427d-4c69-80d5-ae244e8c943a.png)
i didn't get any result.

but when i use Baidu PaddlePaddle Ocr:
https://github.com/PaddlePaddle/PaddleOCR/blob/release/2.6/README_ch.md
![image](https://user-images.githubusercontent.com/537730/205287446-8ea22cfe-318c-41ac-8535-12c40a62c5b6.png)
7 and 4 was detected successfully

so what should I do for next step?

**Comment by gaotongxiao** [comment]
MMOCR supports multiple text detection & recognition algorithms, and the default combination might not be the best to fit certain use cases. You're welcome to experiment with different model combinations by specifying `--det` and `--recog`. See https://mmocr.readthedocs.io/en/latest/demo.html#models for the full model list.

**Comment by melgino** [comment]
I try:
`python mmocr/utils/ocr.py D:/screenshot/222.jpg  --det PS_CTW --recog SAR --kie SDMGR --print-result --imshow`
but I got:
![image](https://user-images.githubusercontent.com/537730/205303501-3585781c-dde2-4a45-b2f8-e4b45baabcdb.png)
how to solve it? 他不应该是自动把预训练模型下载完么? 这报错咋整? 识别和检测的模型，难道我自己先把上述问题解决，再自己去排列组合测试? 没个建议么？我感觉我去除背景噪点后，是一个非常典型，纯粹的ocr识别了啊?

**Reply by gaotongxiao** [reply]
You don't need to specify `kie` if you just want the words from the image. A typical OCR task only consists of text detection and text recognition stage. 

The provided models are trained on academic datasets and may not be general enough to work on every single case. It seems the default text detection model doesn't work well on your image. You can try out different detection models such as `DB_r18` and `DB_r50`.

BTW, the error you encountered was a design flaw in MMOCR 0.x, and you will have to download `wildreceipt`(https://download.openmmlab.com/mmocr/data/wildreceipt.tar) dataset and place it to `data/` in order to use kie model. It has been fixed in MMOCR 1.0, which is now available at `1.x` branch.
